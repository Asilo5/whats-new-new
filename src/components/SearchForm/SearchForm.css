form {
    text-align: center;
    margin: 10px;
}

form > input {
   font-size: 20px;
   border: none;
   border-bottom: 1px solid black;
   padding: 10px;
   width: 20%;
}

form > button {
  font-size: 20px;
  border: none;
  cursor: pointer;
  margin-left: 10px;
}