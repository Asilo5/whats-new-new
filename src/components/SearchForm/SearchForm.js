import React, { Component } from 'react';
import './SearchForm.css';
import PropTypes from 'prop-types';

class SearchForm extends Component {
    constructor() {
      super();
      this.state = {
        newsSearch: '' 
      }
    }

    handleChange = (e) => {
      this.setState({ newsSearch : e.target.value })
    }

    updateState = (e) => {
      e.preventDefault();
      this.props.searchNews(this.state.newsSearch);
      this.clearInputs();
    }

    clearInputs = () => {
      this.setState({ newsSearch : '' })
    }

  render() {
    return (
      <form>
          <input placeholder='Search Article For Here' 
                 name='newsSearch'
                 value={this.state.newsSearch}
                 onChange={this.handleChange}/>
          <button onClick={this.updateState}>Search</button>
      </form>
    )
  }
}

SearchForm.propTypes = {
  searchNews: PropTypes.func
}

export default SearchForm;