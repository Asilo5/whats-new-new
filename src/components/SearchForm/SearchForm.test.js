import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
    
    let mockSearchNews = jest.fn();

    it('should match snapshot with all information passing in correctly', () => {
        let wrapper = shallow(<SearchForm searchNews={mockSearchNews} />);

        expect(wrapper).toMatchSnapshot();
    })
})