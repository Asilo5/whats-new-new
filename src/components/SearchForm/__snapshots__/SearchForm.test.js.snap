// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot with all information passing in correctly 1`] = `
<form>
  <input
    name="newsSearch"
    onChange={[Function]}
    placeholder="Search Article For Here"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
